app/src/main/res/layout/add_profile_activity.xml

```diff
@@ -33,214 +33,215 @@
-            android:contentDescription="@string/current_profile_picture_content_description"
-            android:focusable="true"
-            android:src="@drawable/ic_default_avatar"
-            app:add_shadow="true"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent"
-            app:shadow_radius="8" />
-
-          <ImageView
-            android:id="@+id/add_profile_activity_edit_user_image_view"
-            android:layout_width="48dp"
-            android:layout_height="48dp"
-            android:contentDescription="@string/edit_profile_picture_content_description"
-            android:paddingStart="12dp"
-            android:paddingTop="12dp"
-            android:src="@drawable/rounded_white_background_with_shadow"
-            app:layout_constraintBottom_toBottomOf="@+id/add_profile_activity_user_image_view"
-            app:layout_constraintEnd_toEndOf="@+id/add_profile_activity_user_image_view"
-            app:layout_constraintRight_toRightOf="@+id/add_profile_activity_user_image_view" />
-
-          <TextView
-            android:id="@+id/add_profile_activity_required_heading_text_view"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginStart="28dp"
-            android:layout_marginTop="32dp"
-            android:layout_marginEnd="32dp"
-            android:contentDescription="@string/required_fields_content_description"
-            android:text="@string/add_profile_required"
-            android:textColor="@color/oppiaPrimaryText"
-            android:textSize="12sp"
-            app:layout_constraintTop_toBottomOf="@+id/add_profile_activity_user_image_view" />
-
-          <com.google.android.material.textfield.TextInputLayout
-            android:id="@+id/add_profile_activity_user_name"
-            style="@style/TextInputLayout"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginStart="28dp"
-            android:layout_marginTop="16dp"
-            android:layout_marginEnd="28dp"
-            android:hint="@string/add_profile_input_name"
-            app:errorMessage="@{viewModel.nameErrorMsg}"
+          android:layout_gravity="center_horizontal"
+          android:paddingTop="@dimen/add_profile_main_padding_top"
+          android:paddingBottom="@dimen/add_profile_main_padding_bottom">
+
+          <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="@dimen/add_profile_container_width"
+            android:layout_height="match_parent"
+            android:layout_gravity="center_horizontal"
+            android:paddingStart="@dimen/add_profile_main_padding_start"
+            android:paddingEnd="@dimen/add_profile_main_padding_end"
             app:layout_constraintEnd_toEndOf="parent"
```

Comment:
Add `app:layout_constraintBottom_toBottomOf="parent"`

to fix the test cases failure.

After adding this run all tests, if they fail and run again as sometimes it might not reflect directly and after that it will pass.

I don't know the exact reason but I tried to re-write this code locally without this code and this particular constraint was only difference between the two codes and than I checked in to your code again and I added this constraint and all tests are passing now.

<img width="1509" alt="Screenshot 2021-06-17 at 11 41 47 PM" src="https://user-images.githubusercontent.com/9396084/122451287-96c60980-cfc5-11eb-9d96-add694d7a240.png">

Reply:
Didn't seem to work..

Reply:
@viktoriias Yes, I tried multiple times again and it is still failing.

I think for now we can only minimise this code to separate mobile and tablet files.